### ISSUE TITLE
Windows Container Docker image build fails on step 39/54 with EOF while copying opencv

### ISSUE THREAD

**Issue by NNskelly** [homuler/MediaPipeUnityPlugin#553]
### Plugin Version or Commit ID

v0.9.1

### Unity Version

N/A

### Your Host OS

Windows 10 Pro 21H1 build 19043.1645

### Target Platform

UnityEditor, Android

### [Windows Only] Visual Studio C++ and Windows SDK Version

Visual Studio C++: 16.11.7, 15.9.41, build tools 15.9.36

### [Android Only] Android Build Tools and NDK Version

Should not make a difference yet, but host system has Android Studio managed
Build Tools 28.0.3, 30.0.2, 30.0.3, 31.0.0, 32.0.0
NDK 24.0.8215888

### Command Sequences

following https://github.com/homuler/MediaPipeUnityPlugin/wiki/Installation-Guide#windows verbatim; comparable results from clean checkouts using Command Prompt, PowerShell and Administrator PowerShell.
Docker Desktop is 4.7.1 (77678)
I happen to have host system 
- opencv 3.4.16 under C:\opencv
- Python 3.8.x under C:\Python38
- Python 3.9.x under C:\Python39
- MSYS2 20220503
- bazel 5.1.1
I likely have NumPy and plausibly NuGet squirreled away somewhere from other projects, but my impression of the purpose of Docker is that it ought to containerize/sandbox its own versions of all of the above, notwithstanding the fact that installing a NDK version on the host machine did eliminate a prior error failing to delete an Android temp directory.

`git clone https://github.com/homuler/MediaPipeUnityPlugin.git`
`cd MediaPipeUnityPlugin`
`docker build -t mediapipe_unity:windows . -f docker/windows/x86_64/Dockerfile`

### Log

[plugin build log.txt](https://github.com/homuler/MediaPipeUnityPlugin/files/8636030/plugin.build.log.txt)

### Additional Context

Granted, as noted earlier, some other weird file deletion errors went away when I installed NDK on the host system, so maybe the purpose of Docker Windows Containerization _isn't_ to actually isolate my system from the project context and vice versa and one really needs to follow all the direct Windows 10 install steps before starting the Docker steps even though this is never spelled out in the guide.  But the host system very much does at least have opencv installed under C:\opencv.  If anything, the host system does not have Python specifically under C:\python, but that step appears to have succeeded, unless these steps are running on parallel threads and hitting the command window out of order.  Moreover, unlike the callout for pointing the raw native build process at a differently-located/versioned opencv, I see nothing regarding pointing at a differently-located python installation.

**Comment by homuler** [comment]
> so maybe the purpose of Docker Windows Containerization isn't to actually isolate my system from the project context and vice versa and one really needs to follow all the direct Windows 10 install steps before starting the Docker steps even though this is never spelled out in the guide.

That's just your superstition. You don't need to install NDK, OpenCV, or Python on your host.

> failed to copy files: failed to copy directory: unexpected EOF

Is it possible that you're running out of disk space?

**Comment by NNskelly** [comment]
Thanks for the quick reply.

C drive has 513GB free, D drive has 3.03TB free.  Unless something somewhere is trying to co-opt my USB-tethered phone as extra disk space, I doubt it's a host storage issue.

Independently, if NDK isn't required, then my earlier errors are even more suspicious.  I'm not seeing them anymore, but I was literally 80% of the way to reporting breakage on the NDK install step when the bug report form asked for NDK version and I went ahead and checked what the host had only to discover the host didn't have NDK installed, at which point I installed NDK and, with that being the only change made to the system, for the first time in 3-4 attempts, I did NOT have the docker process break there with a "failed to delete C:\Android\.temp\PackageContents01 in 10 attempts" error in a java step.  I'm kicking myself for blowing away the process copypasta on that failure, so I can't even say that was the actual verbatim path, but what I remember of the traceback suggested the directory may still have been in use, so it might have been another manifestation of the file deletion error that blocks MSYS2, which I think I also encountered when my very first attempt was installing p7zip.

**Comment by homuler** [comment]
>  I'm not seeing them anymore

So may I close the issue?
I don't know why this problem occurred, but if you no longer encounter it either, there's no way to investigate the cause.
In any case, there is no need to install the NDK in the host environment to use Docker images.
I don't deny that installing the NDK may have solved your problem, though.

If you're still having trouble with copying OpenCV, will you try building the image without cache again? (run `docker build --no-cache -t mediapipe_unity:windows . -f docker/windows/x86_64/Dockerfile`)

**Comment by NNskelly** [comment]
_NDK file deletion issue_ seemed to have resolved itself, even after uninstalling NDK from the host for giggles.
opencv issue still present as of last post.  Trying a new fresh checkout and a build with no cache, as directed; seeing some output I don't remember from any prior attempt.  Will post results when it completes or fails.

**Comment by NNskelly** [comment]
alright- cache-bust rebuild on a clean checkout in plain old Command Prompt seems to have resolved everything.  Not sure what state the first few runs got into or why; only difference was I started those in PowerShell and eventually tried in Admin PowerShell to see if that might help with the deletion.

**Comment by NNskelly** [comment]
progressed to setting up inside the container; looks like the bazel_android_fixes.diff is failing with a "patch does not apply" error on two files:
```
C:\bazel>git apply ..\mediapipe\third_party\bazel_android_fixes.diff
error: patch failed: src/main/java/com/google/devtools/build/lib/bazel/rules/android/android_ndk_cc_toolchain_template.t
xt:14
error: src/main/java/com/google/devtools/build/lib/bazel/rules/android/android_ndk_cc_toolchain_template.txt: patch does
 not apply
error: patch failed: src/main/java/com/google/devtools/build/lib/rules/cpp/CppFileTypes.java:31
error: src/main/java/com/google/devtools/build/lib/rules/cpp/CppFileTypes.java: patch does not apply
```

New issue?  Continue here?  Non-issue?  Got a project slack or discord to help suss out if ongoing turbulence is a proper bug or just a usage oversight?

**Comment by homuler** [comment]
It seems to me that you're applying the patch twice.

If you don't know what's happening, use Docker Linux Container to build android libraries.
https://github.com/homuler/MediaPipeUnityPlugin/wiki/Installation-Guide#docker-linux-container

Building AAR on Windows Container is experimental and the procedure is relatively complex and error-prone.

**Comment by NNskelly** [comment]
Thanks.  It doesn't look like the patch applied at all when running the automated git process, and I was too lazy to do a deep dive and figure out if the issue was e.g. checksums, pathing, or whatever else.  Ultimately, I examined the diff file, copied the 2 relevant files out of the container, applied the changes by hand, and copied them back in.  The rest of the steps went smoothly on the fresh Windows container, excepting that after building several targets, I needed to `python build.py clean` to free up space.  It looks like a 20GB default-sized container isn't quite enough for all permutations of all intermediaries.
